src/features/payment/components/PaymentChoice/PaymentChoice.tsx

```diff
@@ -1,110 +1,112 @@
 import {
   SessionItems,
   setSessionItem,
 } from "../../../../utils/storage/sessionStorage";
-import { PaymentInstruments } from "../../models/paymentModel";
-import { DisabledPaymentMethods, EnabledPaymentMethods } from "./PaymentMethod";
-
-function groupByTypeCode(array: Array<PaymentInstruments>) {
-  return array.reduce((acc, current) => {
-    if (!acc[current.paymentTypeCode]) {
-      acc[current.paymentTypeCode] = [];
-    }
-
-    acc[current.paymentTypeCode].push(current);
-    return acc;
-  }, {} as Record<TransactionMethods, Array<PaymentInstruments>>);
-}
+import {
+  PaymentCodeType,
+  PaymentCodeTypeEnum,
+  PaymentInstrumentsType,
+} from "../../models/paymentModel";
+import { PAYMENT_METHODS_CHOICE } from "../../../../utils/config/mixpanelDefs";
+import { DisabledPaymentMethods, MethodComponentList } from "./PaymentMethod";
 
-function getSortedPaymentMethods(
-  groupedMethods: Record<TransactionMethods, Array<PaymentInstruments>>
-) {
-  const paymentMethods: Array<PaymentInstruments> = [];
-  const methodCP = groupedMethods[TransactionMethods.CP]?.[0];
-  const methodCC = groupedMethods[TransactionMethods.CC]?.[0];
+const shouldBeFirst = (method: PaymentInstrumentsType) =>
+  method.paymentTypeCode === PaymentCodeTypeEnum.CP;
 
-  for (const key in groupedMethods) {
-    if (
-      key !== TransactionMethods.CP &&
-      key !== TransactionMethods.CC &&
-      PaymentMethodRoutes[key as TransactionMethods]
-    ) {
-      paymentMethods.push(groupedMethods[key as TransactionMethods][0]);
-    }
+const sortMethods = (a: PaymentInstrumentsType, b: PaymentInstrumentsType) => {
+  if (shouldBeFirst(a)) {
+    return -1;
+  } else if (shouldBeFirst(b)) {
+    return 1;
   }
+  return a.name.localeCompare(b.name);
+};
 
-  const sortedMethods = paymentMethods.sort((a, b) =>
-    a.label.localeCompare(b.label)
+const getNormalizedMethods = (
+  paymentInstruments: Array<PaymentInstrumentsType>
+) => {
+  const { methods, duplicatedMethods } = paymentInstruments.reduce<{
+    foundTypes: Array<PaymentCodeType>;
+    methods: Array<PaymentInstrumentsType>;
+    duplicatedMethods: Array<PaymentInstrumentsType>;
+  }>(
+    (acc, method) =>
+      acc.foundTypes.includes(method.paymentTypeCode)
+        ? { ...acc, duplicatedMethods: acc.duplicatedMethods.concat(method) }
+        : {
+            ...acc,
+            methods: acc.methods.concat(method),
+            foundTypes: acc.foundTypes.concat(method.paymentTypeCode),
+          },
+    {
+      foundTypes: [],
+      methods: [],
+      duplicatedMethods: [],
+    }
   );
 
-  methodCC && sortedMethods.unshift(methodCC);
-  methodCP && sortedMethods.unshift(methodCP);
-
-  return sortedMethods;
-}
+  const { enabled, disabled } = methods.reduce<{
+    enabled: Array<PaymentInstrumentsType>;
+    disabled: Array<PaymentInstrumentsType>;
+  }>(
+    (acc, method) =>
+      method.status === "ENABLED"
```

Comment:
The ecommerce generated code contains a StatusEnum for not use a String "ENABLED"?

Reply:
You are right, thanks. Fixed with the latest commit